lib/v2/screens/profile_screens/profile/components/profile_bottom.dart

```diff
@@ -30,109 +28,11 @@ class ProfileBottom extends StatelessWidget {
-                                          shape: BoxShape.circle,
-                                          color: AppColors.white,
-                                        ),
-                                        child: Center(
-                                          child: Text(
-                                            'to'.i18n,
-                                            style: Theme.of(context)
-                                                .textTheme
-                                                .headline7
-                                                .copyWith(color: AppColors.primary),
-                                          ),
-                                        ),
-                                      ),
-                                    ),
-                                    Flexible(
-                                      child: Text('Citizen'.i18n, style: Theme.of(context).textTheme.headline6Green),
-                                    ),
-                                  ],
-                                ),
-                              ),
-                              Row(
-                                children: [
-                                  MaterialButton(
-                                    shape: RoundedRectangleBorder(
-                                      borderRadius: BorderRadius.circular(8.0),
-                                    ),
-                                    color: AppColors.green1,
-                                    padding: const EdgeInsets.all(8.0),
-                                    onPressed: () => NavigationService.of(context).navigateTo(
-                                      Routes.citizenship,
-                                      ProfileValuesArguments(
-                                        profile: BlocProvider.of<ProfileBloc>(context).state.profile!,
-                                        scores: BlocProvider.of<ProfileBloc>(context).state.score!,
-                                      ),
-                                    ),
-                                    child: Text(
-                                      'View your progress'.i18n,
-                                      style: Theme.of(context).textTheme.subtitle3,
-                                    ),
-                                  ),
-                                ],
-                              ),
-                            ],
-                          ),
-                        ),
-                      ),
-                    ],
-                  ),
-                ],
-              ),
+            BlocBuilder<ProfileBloc, ProfileState>(
+              buildWhen: (previous, current) => previous.profile != current.profile,
+              builder: (context, state) {
+                return state.profile != null ? CitizenshipCard(profile: state.profile!) : const SizedBox.shrink();
+              },
```

Comment:
And from the previous comment, here you only called CitizenshipCard with a const constructor and without having to go through the profile in the constructor.

`const CitizenshipCard(),`

Reply:
Fix